File: python/src/coding_interview/domain/stock_symbol.py
from __future__ import annotations

from enum import Enum


class StockSymbol(Enum):
    """銘柄を表す。"""

    Toyopa = "Toyopa"
    Somy = "Somy"

    @classmethod
    def from_string(cls, s: str) -> "StockSymbol | None":
        try:
            return cls(s)
        except ValueError:
            return None

    def __str__(self) -> str:
        return self.value

File: ruby/lib/coding_interview/domain/stock_symbol.rb
module CodingInterview
  module Domain
    # 銘柄を表す。
    module StockSymbol
      Toyopa = :Toyopa
      Somy = :Somy

      ALL = [Toyopa, Somy].freeze

      def self.from_string(s)
        case s
        when "Toyopa" then Toyopa
        when "Somy" then Somy
        end
      end
    end
  end
end
